### PR TITLE
chore: Make NodeTasks less failure prone

### LINKED ISSUE #9839: Cleanup: Rework `NodeTasks(Builder builder)`
NodeTasks command creation should be checked through so it's more readable and not have NPE possibilities in command creation.

Tasks with many constructor parameters should be made easier to construct.

### PR DESCRIPTION
**Pull request by caalador**
Re-order some of the commands
so they don't by mistake fail due to
missing initializations.

Clean un-used code.

Add checks to not add commands
if missing required information.

Update old JavaDocs

Closes #9839

### CODE REVIEW

File: flow-maven-plugin/src/main/java/com/vaadin/flow/plugin/maven/BuildFrontendMojo.java [hunk 1]
```diff
@@ -136,13 +136,6 @@
     @Parameter(defaultValue = FrontendUtils.WEBPACK_GENERATED)
     private String webpackGeneratedTemplate;
 
-    /**
-     * Copy the `sw.ts` from the specified URL. Default is the template provided
-     * by this plugin. Set it to empty string to disable the feature.
-     */
-    @Parameter(defaultValue = FrontendUtils.SERVICE_WORKER_SRC)
-    private String serviceWorkerTemplate;
```
Comment (mshabarov) [inline flow-maven-plugin/src/main/java/com/vaadin/flow/plugin/maven/BuildFrontendMojo.java:144]: Isn't this really needed anymore?

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeTasks.java [hunk 2]
```diff
@@ -604,7 +602,7 @@ private NodeTasks(Builder builder) {
                             builder.tokenFileData, builder.enablePnpm));
 
             commands.add(new TaskUpdateThemeImport(builder.npmFolder,
-                frontendDependencies.getThemeDefinition(), // NOSONAR
+                frontendDependencies == null ? null : frontendDependencies.getThemeDefinition(),
```
[the remark this answers is not in the document]
Comment (caalador) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeTasks.java]: I've missed that as I jumped around.

File: flow-maven-plugin/src/main/java/com/vaadin/flow/plugin/maven/BuildFrontendMojo.java (hunk 1 above)
Comment (mshabarov) [reply flow-maven-plugin/src/main/java/com/vaadin/flow/plugin/maven/BuildFrontendMojo.java:144]: Yep, I see it now, thanks!

**Review by mshabarov — DISMISSED** [review]

**Comment by mshabarov** [comment]
`BuildFrontendMojoTest` still contains `serviceWorkerTemplate` parameter and, thus, fails.

**Comment by caalador** [comment]
Fixed. Good thing the builds work again.

**Review by mshabarov — APPROVED** [review]

**Pull request merged**